### PR TITLE
fix: Nested GridTable leaf cards draw their own borders

### PR DESCRIPTION
**Pull request by bdow**

### CODE REVIEW

**Review by chr1sjf0x — APPROVED** [review]
🍣

File: src/components/Table/GridTable.tsx [hunk 1]
```diff
@@ -462,47 +451,17 @@ function renderCssGrid<R extends Kinded>(
...
-    Ex: When we have nestedCard styled headers, then we have opening and closing
-    Chrome rows. Therefore, the first non-header row is the fourth row because:
-    - Row 1 is the opening Chrome Row
-    - Row 2 is the header
-    - Row 3 is the closing Chrome Row
-    - Row 4 is the first non-header row
-    And since CSS selectors are not zero-based, we are aiming for a value of 4
-    (3 + 1).
-  */
-  const firstNonHeaderRowIndex = (!isNestedCardStyleHeader ? 1 : 3) + 1;
```
[the remark this answers is not in the document]
Comment (bdow) [reply src/components/Table/GridTable.tsx:487]: Prior to this change the header consisted of 3 rows - two chrome rows and the GridRow. Now there is only the Grid Row. Regardless of using nested card styles, the first non-header row is always the second element.

File: src/components/Table/GridTable.tsx [hunk 2]
```diff
@@ -336,48 +333,40 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
         );
       // Even if we don't pass the filter, one of our children might, so we continue on after this check
       let isCard = false;
+      const isLeaf = row.children === undefined;
```
[the remark this answers is not in the document]
Comment (bdow) [reply src/components/Table/GridTable.tsx:336]: Good question... will think about it.

File: src/components/Table/GridTable.tsx [hunk 3]
```diff
@@ -322,9 +321,7 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
     const filteredRows: RowTuple<R>[] = [];
 
     // Misc state to track our nested card-ification, i.e. interleaved actual rows + chrome rows
-    // for the header and row (non-header rows) cards.
-    const headerCards = !!style.nestedCards && new NestedCards(columns, headerRows, style.nestedCards);
-    const rowCards = !!style.nestedCards && new NestedCards(columns, filteredRows, style.nestedCards);
+    const nestedCards = !!style.nestedCards && new NestedCards(columns, filteredRows, style.nestedCards);
```
Comment (KoltonG) [inline src/components/Table/GridTable.tsx:324]: **praise:** Nice that we no longer need this anymore!

File: src/components/Table/GridTable.tsx [hunk 4]
```diff
@@ -336,48 +333,40 @@ export function GridTable<R extends Kinded, S = {}, X extends Only<GridTableXss,
...
-          // This will determine if we should include opening and closing
-          // Chrome rows.
-          const isHeaderNested = !!style.nestedCards?.kinds["header"];
-
-          isHeaderNested && headerCards && headerCards.maybeOpenCard(row);
-          headerRows.push([row, makeRowComponent(row)]);
-          isHeaderNested && headerCards && headerCards.closeCard();
-
-          isHeaderNested && headerCards && headerCards.done();
-
```
Comment (KoltonG) [inline src/components/Table/GridTable.tsx:368]: **praise:** It's nice that we no longer need this!

File: src/components/Table/GridTable.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (chr1sjf0x) [reply src/components/Table/GridTable.tsx:487]: TY for the explanation.

I am curious as to whether this unwinds any of @KoltonG 's recent work regarding the spacing at the top of the virtualized list. And whether this has had an impact on perf.

File: src/components/Table/GridTable.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (bdow) [reply src/components/Table/GridTable.tsx:336]: Makes sense to me. I'll add that in.

File: src/components/Table/GridTable.tsx [hunk 5]
```diff
@@ -1018,6 +980,7 @@ function GridRow<R extends Kinded, S>(props: GridRowProps<R, S>): ReactElement {
         maybeAddCardPadding(openCardStyles, "first", {
           ...maybeStickyHeaderStyles,
           ...(columnSizes ? Css.w(columnSizes[0]).$ : {}),
+          ...(openCardStyles.length === 0 ? getLeafCardStyles(leafCardStyles, "first") : {}),
```
Comment (KoltonG) [inline src/components/Table/GridTable.tsx:983]: **question:** What does `openCardStyles` signify again in this instance? Is this `0` when it's a leaf node?

File: src/components/Table/GridTable.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (bdow) [reply src/components/Table/GridTable.tsx:487]: It simplifies @KoltonG's work. The spacing remains.

There is more work to do in this PR that was discovered as part of the Storybook UI tests. Once done I will validate performance changes against Kolton's [last PR where he included metrics](https://github.com/homebound-team/beam/pull/409).

**Review by KoltonG — APPROVED** [review]
The code looks good and there's a lot less complicated headerCard stuff since we are no longer doing Chrome rows for the header! Thank you! Heads up on the Chromatic diffs!

**Comment by bdow** [comment]
The approach of this has completely changed. I am going to close out this PR and open a new one. Thanks for the reviews and feedback here, though!

**Comment by bdow** [comment]
Moved here: https://github.com/homebound-team/beam/pull/444

**Pull request closed without merging**